README.md

```diff
@@ -137,6 +137,42 @@ ann_id = omero2pandas.upload_table(my_data, "Name for table", 142, "Image")
 Once uploaded, the table will be accessible on OMERO.web under the file 
 annotations panel of the parent object. Using unique table names is advised.
 
+### Large Tables
+The first argument to `upload_table` can be a pandas dataframe or a path to a 
+.csv file containing the table data. In the latter case the table will be read 
+in chunks corresponding to the `chunk_size` argument. This will allow you to 
+upload tables which are too large to load into system memory.
+
+```python
+import omero2pandas
+ann_id = omero2pandas.upload_table("/path/to/my.csv", "My table", 
+                                   142, chunk_size=100)
+# Reads and uploads the file to Image 142, loading 100 lines at a time 
+```
+
+
+### Linking to additional objects
+
+Each table gets linked to a parent object determined by the `parent_id` and
+`parent_id` parameters. This becomes the main parent that the table is 
+associated with in OMERO. You can also supply the `extra_links` parameter to 
+provide extra objects to link the table to. This should be a list of 
+tuples in the format `(<target_type>, <target_id>)`.
+
+
+```python
+import omero2pandas
+ann_id = omero2pandas.upload_table(
+    "/path/to/my.csv", "My table", 142,
+    extra_links=[("Image", 101), ("Dataset", 2), ("Roi", 1923)])
```

Comment:
I think you're right that we could slim this down significantly by just doing all the links as a `links` argument and dropping parent_type/parent_id. To maintain backwards compatibility perhaps we keep those arguments but deprecate them.

I had kept the idea of a "primary" parent as the one we validate before trying to attach a FileAnnotation to it, but perhaps we just check that all the provided links are visible/writeable before trying to upload the table?

Reply:
Agreed, if that solution works for everyone and even if this component is sub 1.0.0, I would vote for going through a proper round of deprecation of `parent_type/parent_id` and internally handling them alongside the `links` tuple.

On the primary parent, I guess it's really an end-user question. The reality is that there are multiple scenarios where a table might be linked to multiple objects. ROI tables linked to both images and ROIs as described in https://www.glencoesoftware.com/blog/2024/02/28/whats-new-in-omero-plus.html is one of one.

I would err in favor of implementing the same validation logic and outcome for all parent objects for simplicity. This have the benefit of prevent scenarios where a table is created and only partially linked as we know it can be tricky to troubleshoot down the line.